src/builder.rs

```diff
@@ -483,6 +495,29 @@ impl SessionBuilder {
     }
 }
 
+/// Specifies how long the controlling ssh process should stay alive.
+#[derive(Clone, Debug, Default)]
+pub enum ControlPersist {
+    /// Will stay alive indefinitely.
+    #[default]
+    Forever,
+    /// Will be closed after the initial connection is closed
+    ClosedAfterInitialConnection,
+    /// If the ssh control server has been idle for specified duration, it will
+    /// exit.
```

Comment:
```suggestion
    /// If the ssh control server has been idle for specified duration, it will
    /// exit.
    ///
    /// If you specify a value less than 1 second, then it would be treated as expiring
    /// immediately if the control server is idle.
```

Reply:
According to the ssh_config docs, passing `0` (or `0s`) is the same as passing `yes`. I've tried this manually just now and the session sticks around forever.

If we want the behavior you describe, we can either set set a minimum of 1 second.

I think putting a requirement of 1 second is preferable rather than having it stick around forever -- if someone uses `IdleFor` they will expect the session to eventually go away.

```suggestion
    /// If the ssh control server has been idle for specified duration, it will
    /// exit.
    ///
    /// If you specify a value less than 1 second, it will be rounded up to
    /// 1 second (the minimum possible time)
```

Reply:
Yeah that sounds more reasonable.

We should also warn user that if they pass a duration set to 0, then it would be the same as Forever.

Reply:
Or, we could replace Duration with NonZeroUsize, representing number of seconds, which is less confusing.

What do you think?

Reply:
cc @krfricke just in case you missed my comment

Reply:
I think either is fine (Duration that is rounded to minimum of 1 second, or NonZeroUsize). Both are usually constructed from seconds as a unit, and converting between them is also pretty straightforward.

But I do like the semantics of NonZeroUsize, as it avoids the rounding and the millisecond precision of duration is ignored anyway. So I'll update the PR to use that instead